lib/intercept.js

```diff
@@ -298,12 +307,73 @@ function overrideClientRequest() {
       }
     }
   }
-  inherits(OverriddenClientRequest, http.ClientRequest)
+  function OverriddenClientRequest(input, options, cb) {
```

Comment:
There is a lot of duplicated code here. If the legacy behavior is conditioned on a constant exported from `common.js`, could this be handled more simply, e.g. with an if statement around the part which differs?